CMakeLists.txt

```diff
@@ -274,30 +278,61 @@ IF(CMAKE_SYSTEM_NAME STREQUAL "Darwin")
     SET_TARGET_PROPERTIES(${PROJECT_TARGET_BIN} PROPERTIES MACOSX_BUNDLE_INFO_PLIST "${CMAKE_SOURCE_DIR}/resources/Info.plist")
 ENDIF()
 
+# Fixes for Visual Studio 2017 immediate build\execution\debug directly opening CMake folder without project
+# Initialize IS_VS_2017CMAKE to 0, it will be override by Visual Studio cmake with -D command line parameter
+SET(IS_VS_2017CMAKE 0 CACHE BOOL "Executed from VS2017 IDE")
```

Comment:
We should prepare config to be used both from cli and gui - not hardcode such configs